### PR TITLE
Switch to using jessepeterson/cfgprofiles for profile generation

### PR DESCRIPTION
**Pull request by jessepeterson**
Switch to https://github.com/jessepeterson/cfgprofiles library for generation of enrollment profile and OTA profiles and payloads.

Minor cleanup to reduce duplication of effort and center Golang profile generation in one place.

Diff between before and after profile generation:

```diff
--- /tmp/p1.plist	2022-06-01 11:44:43.000000000 -0700
+++ /tmp/p2.plist	2022-06-01 11:53:53.000000000 -0700
@@ -43,12 +43,10 @@
         <string>com.github.micromdm.micromdm.enroll.scep</string>
         <key>PayloadOrganization</key>
         <string>MicroMDM</string>
-        <key>PayloadScope</key>
-        <string>System</string>
         <key>PayloadType</key>
         <string>com.apple.security.scep</string>
         <key>PayloadUUID</key>
-        <string>652ad7f2-f655-4c67-984a-b8b43babd3be</string>
+        <string>30985600-2910-417F-B3AD-729B8DD0EB99</string>
         <key>PayloadVersion</key>
         <integer>1</integer>
       </dict>
@@ -59,21 +57,17 @@
         <string>https://mdm.example.com/mdm/checkin</string>
         <key>CheckOutWhenRemoved</key><true/>
         <key>IdentityCertificateUUID</key>
-        <string>652ad7f2-f655-4c67-984a-b8b43babd3be</string>
+        <string>30985600-2910-417F-B3AD-729B8DD0EB99</string>
         <key>PayloadDescription</key>
         <string>Enrolls with the MDM server</string>
-        <key>PayloadDisplayName</key>
-        <string></string>
         <key>PayloadIdentifier</key>
         <string>com.github.micromdm.micromdm.enroll.mdm</string>
         <key>PayloadOrganization</key>
         <string>MicroMDM</string>
-        <key>PayloadScope</key>
-        <string>System</string>
         <key>PayloadType</key>
         <string>com.apple.mdm</string>
         <key>PayloadUUID</key>
-        <string>3d8ecb9a-575b-4b63-b955-e217df0ef108</string>
+        <string>08DDDB53-2D68-4C7B-922B-E5A40EAE4335</string>
         <key>PayloadVersion</key>
         <integer>1</integer>
         <key>ServerCapabilities</key>
@@ -101,7 +95,7 @@
     <key>PayloadType</key>
     <string>Configuration</string>
     <key>PayloadUUID</key>
-    <string>206d254f-1718-4b76-aeac-2eac459989dc</string>
+    <string>47093D57-CC38-43FC-BC1B-4204FFD4BDA0</string>
     <key>PayloadVersion</key>
     <integer>1</integer>
   </dict>
```

Note `PayloadScope` is a profile-level attribute (per docs) and so is removed from the payloads. `PayloadDisplayName` is optional and so is removed. Otherwise the only real changes between the profiles are the UUIDs.

### CODE REVIEW

**Comment by jessepeterson** [comment]
Oops, have to update tests. Shame on me. :)

File: mdm/enroll/service.go [hunk 1]
```diff
@@ -164,148 +179,149 @@ func (svc *service) Enroll(ctx context.Context) (profile.Mobileconfig, error) {
...
-	payloadContent = append(payloadContent, mdmPayloadContent)
+	profile.AddPayload(mdmPayload)
 
 	// Client needs to trust us at this point if we are using a self signed certificate.
 	if len(svc.TLSCert) > 0 {
-		tlsPayload := NewPayload("com.apple.security.pem")
+		tlsPayload := cfgprofiles.NewCertificatePKCS1Payload(EnrollmentProfileId + ".cert.selfsigned")
 		tlsPayload.PayloadDisplayName = "Self-signed TLS certificate for MicroMDM"
 		tlsPayload.PayloadDescription = "Installs the TLS certificate for MicroMDM"
 		tlsPayload.PayloadIdentifier = EnrollmentProfileId + ".cert.selfsigned"
```
Comment (korylprince) [inline mdm/enroll/service.go:249]: This should be redundant now because the ID is set above.

**Review by korylprince — CHANGES_REQUESTED** [review]
I did a cursory overview and don't see any issues (other than the one comment I left).

I haven't been able to actually test this against an actual enrollment; assuming you've done some testing on this.

File: mdm/enroll/service.go (hunk 1 above)
Comment (jessepeterson) [reply mdm/enroll/service.go:249]: Er.. sorry I'm an idiot... which ID is set where? ;) This is for the separate included TLS profile payload in the profile.

Comment (jessepeterson) [reply mdm/enroll/service.go:249]: Oh sorry I see it now I think. :) the duplicated identifier.

Comment (korylprince) [reply mdm/enroll/service.go:249]: I thought it would highlight that particular line :)

Comment (korylprince) [reply mdm/enroll/service.go:249]: But, yeah. The `cfgprofiles.NewCertificatePKCS1Payload` sets the ID, then `tlsPayload.PayloadIdentifier = EnrollmentProfileId + ".cert.selfsigned"` sets it again.

**Review by korylprince — APPROVED** [review]

**Pull request merged**